src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/TickerTask.java

```diff
@@ -215,12 +341,28 @@ public void halt() {
 
     @ParametersAreNonnullByDefault
     public void queueMove(Location from, Location to) {
-        movingQueue.put(from, to);
+        // This collection is iterated over in a different thread. Need to lock it.
+        new Throwable().printStackTrace();
+        queueLock.readLock().lock();
+        try {
+            movingQueue.put(from, to);
+        }
+        finally {
+            queueLock.readLock().unlock();
+        }
     }
 
     @ParametersAreNonnullByDefault
     public void queueDelete(Location l, boolean destroy) {
-        deletionQueue.put(l, destroy);
+        // This collection is iterated over in a different thread. Need to lock it.
+        new Throwable().printStackTrace();
```

Comment:
Why are you printing a stacktrace here...?

Reply:
Oops that's from debugging